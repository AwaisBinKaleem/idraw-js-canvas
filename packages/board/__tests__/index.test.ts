import Board from './../src';

test('testing', async () => {  
  document.body.innerHTML = `
    <div id="mount"></div>
  `;
  const opts = {
    width: 600,
    height: 400,
    contextWidth: 600,
    contextHeight: 400,
    devicePixelRatio: 4
  }
  const mount = document.querySelector('#mount') as HTMLDivElement;
  const board = new Board(mount, opts);

  const ctx = board.getContext();
  const data = {
    elements: [
      {
        x: 10,
        y: 10,
        w: 200,
        h: 120,
        type: 'rect',
        desc: {
          color: '#f0f0f0',
        }
      },
      {
        x: 80,
        y: 80,
        w: 200,
        h: 120,
        type: 'rect',
        desc: {
          color: '#cccccc',
        }
      },
      {
        x: 160,
        y: 160,
        w: 200,
        h: 120,
        type: 'rect',
        desc: {
          color: '#c0c0c0',
        }
      },
      {
        x: 400 - 10,
        y: 300 - 10,
        w: 200,
        h: 100,
        type: 'rect',
        desc: {
          color: '#e0e0e0',
        }
      }
    ]
  };

  board.clear();
  ctx.clearRect(0, 0, opts.width, opts.height);
  ctx.setFillStyle('#ffffff');
  ctx.fillRect(0, 0, opts.width, opts.height);
  data.elements.forEach(ele => {
    ctx.setFillStyle(ele.desc.color);
    ctx.fillRect(ele.x, ele.y, ele.w, ele.h);
  });
  board.draw();

  const originCtx = board.getOriginContext();
  // @ts-ignore;
  const originCalls = originCtx.__getDrawCalls();
  expect(originCalls).toMatchSnapshot();

  const displayCtx = board.getDisplayContext();
  // @ts-ignore;
  const displayCalls = displayCtx.__getDrawCalls();
  expect(displayCalls).toMatchSnapshot();

});