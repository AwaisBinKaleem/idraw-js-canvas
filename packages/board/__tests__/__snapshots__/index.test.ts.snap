// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing 1`] = `
Array [
  Object {
    "props": Object {
      "height": 1600,
      "width": 2400,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {
      "height": 1600,
      "width": 2400,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
]
`;

exports[`testing 2`] = `
Array [
  Object {
    "props": Object {
      "height": 1600,
      "width": 2400,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  Object {
    "props": Object {
      "height": 1600,
      "width": 2400,
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
]
`;
